lib/DocumentManagerInterface.php

```diff
@@ -122,4 +124,56 @@ public function createQuery($query, $locale = null, array $options = []);
      * http://doctrine-phpcr-odm.readthedocs.org/en/latest/reference/query-builder.html
      */
     public function createQueryBuilder();
+
+    /**
+     * Retrieve the document inspector.
+     *
+     * NOTE: In the future this will be a factory for retrieving inspectors
+     * for a given object/class.
+     *
+     * @return DocumentInspector
+     */
+    public function getInspector();
+
+    /**
+     * Return the node manager instance.
+     *
+     * @return NodeManager
+     */
+    public function getNodeManager();
+
+    /**
+     * Return the document registry instance.
+     *
+     * @return DocumentRegistry
+     */
+    public function getRegistry();
+
+    /**
+     * Return the metadata factory instance.
+     *
+     * @return MetadataFactoryInterface
+     */
+    public function getMetadataFactory();
+
+    /**
+     * Return the event dispatcher instance.
+     *
+     * @return EventDispatcher
+     */
+    public function getEventDispatcher();
+
+    /**
+     * Return the PHPCR session instance.
+     *
+     * @return SessionInterface
+     */
+    public function getSession();
+
+    /**
+     * Return the proxy factory instance.
+     *
+     * @return ProxyFactory
+     */
+    public function getProxyFactory();
```

Comment:
This was a more difficult decision. All of these things are scoped to
the document manager instance. They belong to the document manager and
are not shared by anything else (with the exception of the
MetadataFactory, but in the future that might change).

The alternative was the `DocumentContext` which was an accessor for the
services contained within the DocumentManager. But in practice this
seemed like a pointless step.

The Doctrine Entity/Document Managers all provide access to their
various components through their Managers.

On Thu, Apr 14, 2016 at 01:03:32AM -0700, Daniel Rotter wrote:
>    In [1]lib/DocumentManagerInterface.php:
> 
>  > +     */
>  > +    public function getEventDispatcher();
>  > +
>  > +    /**
>  > +     * Return the PHPCR session instance.
>  > +     *
>  > +     * @return SessionInterface
>  > +     */
>  > +    public function getSession();
>  > +
>  > +    /**
>  > +     * Return the proxy factory instance.
>  > +     *
>  > +     * @return ProxyFactory
>  > +     */
>  > +    public function getProxyFactory();
> 
>    Is it really necessary to put all of these into this interface? Until now
>    I thought as the DocumentManager as a facade for a more complex underlying
>    system. This way we populate quite a lot of these complex classes, making
>    later refactorings pretty hard.
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu-document-manager/pull/66#discussion_r59676468
>    2. https://github.com/sulu/sulu-document-manager/pull/66/files/6e7fab9301c97bff800cfff05d913f78e432ded1#r59676468
>    3. https://github.com/sulu/sulu-document-manager/pull/66/files/6e7fab9301c97bff800cfff05d913f78e432ded1#r59676468